### ISSUE TITLE
Unicode is not supported in Octopoes

### ISSUE THREAD

**Issue by originalsouth** [minvws/nl-kat-coordination#2527]
**Describe the bug**
Octopoes does not support Unicode.

**To Reproduce**
For instance: 
Add Network OOI named "🐱".
Retrieve "422 Client Error: Unprocessable Entity for url: http://octopoes_api:80/0/declarations"

Note that URLs -- like for instance http://☮️.com -- *are* supported via Pydantic as they are implicitly converted to Punycode.

**Expected behavior**
OOI is added without error.

**Screenshots**
![image](https://github.com/minvws/nl-kat-coordination/assets/5300799/2ebcf344-d527-42da-9a1d-6a211c478299)

**OpenKAT version**
Current main (3bc5a697a1ffa1b1acf9a16f6250dded2690c533)

**Comment by originalsouth** [comment]
Related to #2525

**Comment by originalsouth** [comment]
See also https://github.com/minvws/nl-kat-coordination/pull/2464

**Comment by originalsouth** [comment]
Experimental branch "[potato-model](https://github.com/minvws/nl-kat-coordination/tree/experiment/potato-model)" overriding the `BaseModel.dump_model_json` as a possible fix.

**Comment by originalsouth** [comment]
@dekkers @noamblitz @ammar92  In ce51c1acca242ccfe22fd5fbf2613e380f7df0e7 of [this branch](https://github.com/minvws/nl-kat-coordination/tree/experiment/potato-model), we get the following error in the connector
```
[CRITICAL] [api] [{'type': 'model_attributes_type', 'loc': ('body',), 'msg': 'Input should be a valid dictionary or object to extract fields from', 'input': '{"ooi":{"object_type":"Network","scan_profile":null,"primary_key":"Network|🐱","name":"🐱"},"valid_time":"2024-02-26T09:53:41.132427Z","method":null,"task_id":null}', 'url': 'https://errors.pydantic.dev/2.6/v/model_attributes_type'}]
```
while running `make -C octopoes itest`. The error seem to come from a Pydantic C library while being processed by FastAPI.

**Comment by dekkers** [comment]
The problem was that you changed the request parameter `data=` to `json=`, but data accepts a string and json accepts a dict that requests will encode as json. So what you send to FastAPI was a JSON-encoded string which pydantic does not understand.

With regards to unicode the problem is that requests does not encode data as utf-8. Adding utf-8 like described on https://stackoverflow.com/questions/77652094/how-to-post-json-data-that-include-unicode-characters-to-fastapi-using-python-re fixed this, but also needs to be done for the requests of octopoes to xtdb.

This is also a good reason for switching to httpx as planned in #2299 because httpx encodes strings as utf-8 by default:
https://www.python-httpx.org/compatibility/#content-encoding

**Comment by originalsouth** [comment]
Thanks @dekkers fixed in #2558. Pending on fix for #2527.
